lib/html_parser.dart

```diff
@@ -24,18 +25,21 @@ typedef CustomRender = dynamic Function(
   dom.Element element,
 );
 
+BuildContext scrollContext;
+
 class HtmlParser extends StatelessWidget {
   final String htmlData;
   final OnTap onLinkTap;
   final OnTap onImageTap;
   final ImageErrorListener onImageError;
   final bool shrinkWrap;
-
+  final ScrollController scrollController = ScrollController();
   final Map<String, Style> style;
   final Map<String, CustomRender> customRender;
   final Map<ImageSourceMatcher, ImageRender> imageRenders;
   final List<String> blacklistedElements;
   final NavigationDelegate navigationDelegateForIframe;
+  final Map<String, BuildContext> buildContexts = {};
```

Comment:
So this is sort of your list of points in the document where you can link to, right? I wonder though if we could use Flutter's keys instead of maintaining our own list? That would also dramatically reduce the changes required in this PR.

In `StyledElement` we already have the `elementId` it seems. The `StyledText` that we eventually will create for (all? almost all?) tags could set its StatelessWidget key. With this key (which I think can be a ValueKey, as ids shoud be unique in html?) you can retrieve the `currentContext` which give you the context you wanted to scroll to. Or am I missing something?

You wrote that you tried this, but then tried to access the renderobject which was always null. Can't we do `Scrollable.ensureVisible(key.currentContext)`?

Reply:
Yeah I know. It needs to be a `GlobalKey` because otherwise you cannot access `currentContext`. 

I may have wrote incorrectly, but it was actually `currentContext` coming back as null which was more concerning. That makes me think the key wasn't even applying to the widgets somehow?? I used Flutter Inspector to check this and sure enough every widget had a different key ID assigned to it than it should have. I also checked my algorithm by printing the key ID right before the widget was rendered and it did print the correct ID, but still the widget itself had a different key.